src/Shared/StringSplit/StringSplitExtensions.cs

```diff
@@ -0,0 +1,891 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
Do we use **all** the API in this class? 😱

Reply:
I don't think so. What is your suggestion? Leave only minimum necessary set of APIs?

Reply:
Yes, please keep only the necessary API.

Reply:
done